src/controllers/MainController.js

```diff
@@ -0,0 +1,31 @@
+import InputView from '../views/InputView.js';
+import OutputView from '../views/OutputView.js';
+import { InputValidator } from '../models/validations/InputValidator.js';
+import { isEmptyInput } from '../models/validations/isEmptyInput.js';
+import { updateSeparators } from '../utils/updateSeparators.js';
+import { parsingWithSeparators, parsingNumberPart } from '../utils/parser.js';
+import { Calculator } from '../models/Calculator.js';
+
+export class MainController {
+	async run() {
+		try {
+			const rawInput = await InputView.readInput();
+			if (isEmptyInput(rawInput)) return OutputView.printResult(0);
+			InputValidator.validateStrings(rawInput);
+			const separators = updateSeparators(rawInput, [':', ',']);
+			const numbers = parseInput(rawInput);
+			InputValidator.validateNumbers(numbers);
+			const result = Calculator.calculate(numbers, separators);
+			OutputView.printResult(result);
+		} catch (error) {
+			OutputView.printError('[ERROR]');
+			throw error;
+		}
+	}
+}
+
+function parseInput(input) {
```

Comment:
```suggestion
function parseInput(input, seperators) {
```
로 seperators를 받으면 어떨까 싶습니다. 앞으로 뭔가 더 변경사항이 있다고 할 때, 내부에 seperator를 감추는게 좋을지 노출하는게 좋을지 모르겠지만, 현재 코드베이스만 보았을때에는  seperator를 사용하는 곳에서 받아올 수 있는 상황이긴 합니다. 근데 OOP관점에서는 잘모르겠긴하네요!...

Reply:
저두 separators를 받는게 더 좋지 않을까 라는 생각을 많이 했던거같아요 아예 클래스 내부의 메서드로 옮겨버리던가 하는 방법도 고민해봤는데 이 부분은 저도 좀 더 고민이 필요한 지점인 것 같아요! 좋은 지적 감사합니다

Reply:
저도 보면서 이런저런 고민을 하시다가 남은게 지금의 형태이지 않을까 했습니다. 클래스 내부로 가지고 들어오자니 클래스가 이부분에 대한 역할과 책임을 갖는게 맞나싶으셨을 것 같기도하구..ㅎㅎ저도 보면서 엄청 고민하셨겠구나 했습니다.